### PR TITLE
Add custom parameters to token introspection requests

### PR DESCRIPTION
**Pull request by sayembd**
Added support for providing custom parameters to an OAuth 2.0 token introspection request. This is done by explicitly instantiating a NimbusOAuth2TokenIntrospectionClient instance and then setting a custom Converter implementation.

### CODE REVIEW

File: oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/NimbusOAuth2TokenIntrospectionClient.java [hunk 1]
```diff
@@ -112,22 +108,17 @@ public NimbusOAuth2TokenIntrospectionClient(String introspectionUri, RestOperati
...
-		HttpHeaders headers = new HttpHeaders();
-		headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON_UTF8));
-		return headers;
-	}
+	/**
+	 * Sets the {@link Converter} used for converting the OAuth 2.0 access token to a {@link RequestEntity}
+	 * representation of the OAuth 2.0 token introspection request.
+	 *
+	 * @param requestEntityConverter the {@link Converter} used for converting to a {@link RequestEntity} representation
+	 *                                  of the token introspection request
```
Comment (jzheaux) [inline oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/NimbusOAuth2TokenIntrospectionClient.java]: Just a quick cleanup here; please left-justify this

File: oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/NimbusOAuth2TokenIntrospectionClientTests.java [hunk 2]
```diff
@@ -254,6 +255,18 @@ public void constructorWhenRestOperationsIsNullThenIllegalArgumentException() {
 				.isInstanceOf(IllegalArgumentException.class);
 	}
 
+	@Test
+	public void setRequestEntityConverterWhenConverterIsNullThenExceptionIsThrown() {
```
Comment (jzheaux) [inline oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/introspection/NimbusOAuth2TokenIntrospectionClientTests.java]: We should probably have one more test that uses a mock `Converter` to confirm that the setter works.

File: oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/NimbusOAuth2TokenIntrospectionClient.java [hunk 3]
```diff
@@ -100,8 +97,7 @@ public NimbusOAuth2TokenIntrospectionClient(String introspectionUri, RestOperati
 	 */
 	@Override
 	public Map<String, Object> introspect(String token) {
-		TokenIntrospectionSuccessResponse response = Optional.of(token)
-				.map(this::buildRequest)
+		TokenIntrospectionSuccessResponse response = Optional.of(this.requestEntityConverter.convert(token))
```
[the remark this answers is not in the document]
Comment (sayembd) [reply oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/NimbusOAuth2TokenIntrospectionClient.java]: Makes sense from a readability perspective.

**Comment by sayembd** [comment]
Thanks for the review, @jzheaux, I have made the necessary changes.

**Comment by jzheaux** [comment]
Thanks, @sayembd! This is now merged into `master`.

**Pull request merged**